### ISSUE TITLE
[BUG] Problem importing knative CRDs

### ISSUE THREAD

**Issue by pzfreo** [awslabs/cdk8s#129]
**Describe the bug**
I'm trying to import the KNative serving CRD from https://github.com/knative/serving/releases/download/v0.14.0/serving-crds.yaml

I get the following error:
```
TypeError: Cannot read property 'kind' of null
    at /usr/local/lib/node_modules/cdk8s-cli/lib/import/crd.js:81:34
    at Array.filter (<anonymous>)
    at Function.match (/usr/local/lib/node_modules/cdk8s-cli/lib/import/crd.js:81:14)
    at async matchImporter (/usr/local/lib/node_modules/cdk8s-cli/lib/import/dispatch.js:23:17)
    at async Object.importDispatch (/usr/local/lib/node_modules/cdk8s-cli/lib/import/dispatch.js:7:26)
    at async Object.handler (/usr/local/lib/node_modules/cdk8s-cli/bin/cmds/import.js:27:9)
```
There are a couple of other things worth noting. Firstly, there is no following of redirects on download of HTTP resources (so using that URL doesn't work). Secondly, as you can see there is no clear idea of where in the imported CRD the problem lies.

**To Reproduce**
Since redirects don't work, I downloaded the crd like this:
```
wget https://github.com/knative/serving/releases/download/v0.14.0/serving-crds.yaml
```

Here is my cdk8s.yaml
```
language: typescript
app: node main.js
imports:
  - k8s@1.17.0
  - serving-crds.yaml
```

And then I run `cdk8s import`

**Expected behavior**
It to work as the CRD example.

**Additional context**
I've tried a couple of other CRDs and they work fine.

**Comment by eladb** [comment]
thanks for the report. we'll take a look

**Comment by eladb** [comment]
@iliapolo would you be able to take a look at this?

**Comment by campionfellin** [comment]
I can help too - seems like there is a few things going on here.

1. The `.yaml` there ends with `---` and we're not handling that properly. I think the `parseAllDocuments` grabs the last "empty" doc after the `---`. PR here: https://github.com/awslabs/cdk8s/pull/130 to make sure `doc` is defined before casting it to `CustomResourceApiObject` and trying to get the `kind`.

https://github.com/awslabs/cdk8s/blob/2034ec06f0c9fb419a194cbb23a18e2678076d29/packages/cdk8s-cli/lib/import/crd.ts#L107-L109

2. Parsing still fails, because the `knative` CRD you have has fields that we don't:

https://github.com/awslabs/cdk8s/blob/2034ec06f0c9fb419a194cbb23a18e2678076d29/packages/cdk8s-cli/lib/import/crd.ts#L10-L24

(probably just need to update that to handle this)

3. Figuring out the downloading issue.

4. More friendly error handling :)

**Comment by eladb** [comment]
@campionfellin you got it!

**Comment by eladb** [comment]
This is resolved by #132
